js/trbinance.js

```diff
@@ -0,0 +1,1062 @@
+        // const method = 'privateGetOrdersDetail';
+        // const request = {
+        //     'symbol': market['id'],
+        // };
+        // if (since !== undefined) {
+        //     request['startTime'] = since;
+        // }
+        // if (limit !== undefined) {
+        //     request['limit'] = limit;
+        // }
+        // const response = await this[method] (this.extend (request, params));
+        const response = [
+            {
+                'tradeId': '3',
+                'orderId': '2',
+                'symbol': 'ADA_USDT',
+                'price': '0.04398',
+                'qty': '250',
+                'quoteQty': '10.995',
+                'commission': '0.25',
+                'commissionAsset': 'ADA',
+                'isBuyer': 1,
+                'isMaker': 0,
+                'isBestMatch': 1,
+                'time': '1572920872276',
+            },
+        ];
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    parseAccountSpot (response, type = undefined) {
+        const result = {
+            'info': response,
+        };
+        let timestamp = undefined;
+        if ((type === 'spot') || (type === 'margin')) {
+            timestamp = this.safeInteger (response, 'timestamp');
+            // const balances = this.safeValue2 (response, 'data', 'accountAssets', []);
+            const balances = response.data.accountAssets;
+            for (let i = 0; i < balances.length; i++) {
+                const balance = balances[i];
+                const currencyId = this.safeString (balance, 'asset');
+                const code = this.safeCurrencyCode (currencyId);
+                const account = this.account ();
+                account['free'] = this.safeString (balance, 'free');
+                account['used'] = this.safeString (balance, 'locked');
+                result[code] = account;
+            }
+        } else {
+            let balances = response;
+            if (!Array.isArray (response)) {
+                balances = this.safeValue (response, 'assets', []);
+            }
+            for (let i = 0; i < balances.length; i++) {
+                const balance = balances[i];
+                const currencyId = this.safeString (balance, 'asset');
+                const code = this.safeCurrencyCode (currencyId);
+                const account = this.account ();
+                account['free'] = this.safeString (balance, 'availableBalance');
+                account['used'] = this.safeString (balance, 'initialMargin');
+                account['total'] = this.safeString2 (balance, 'marginBalance', 'balance');
+                result[code] = account;
+            }
+        }
+        result['timestamp'] = timestamp;
+        result['datetime'] = this.iso8601 (timestamp);
+        return this.safeBalance (result);
+    }
+
+    async fetchAccountSpot (symbol = undefined, since = undefined, limit = undefined, params = {}) {
```

Comment:
We don't really have a `fetchAccountSpot` unified method anywhere else, but we should implement `fetchBalance` method and a `fetchTradingFees` method, see how that is done in the other exchanges.

- https://github.com/ccxt/ccxt/wiki/Requirements
- https://github.com/ccxt/ccxt/wiki/Manual#querying-account-balance